### PR TITLE
Add help actions guidance on icons

### PR DESCRIPTION
**Pull request by cveigt**
### :pushpin: Summary

This PR adds clarification and improved guidance on icons for help actions.

Preview: https://hds-website-git-cv-icon-docs-update-hashicorp.vercel.app/icons/usage-guidelines#help-actions

### :camera_flash: Screenshots

Screen Shot 2023-06-23 at 4 52 38 PM

### :link: External links

Jira ticket: [HDS-1928](https://hashicorp.atlassian.net/browse/HDS-1928)

***

### 👀 Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-1928]: https://hashicorp.atlassian.net/browse/HDS-1928?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: website/docs/icons/usage-guidelines/partials/guidelines/overview.md [hunk 1]
```diff
@@ -119,8 +119,21 @@ Some icons represent common actions within our products.
...
+
 ![Examples of common help icons](/assets/foundations/flight-icons/help-actions.png =660x*)
 
+!!! Info
+
+**Other actions**
+
+Use `external-link` for any other help action that points to an external URL.
+!!!
+
```
Comment (heatherlarsen) [inline website/docs/icons/usage-guidelines/partials/guidelines/overview.md:136]: Curious...why add this banner? The `external-link` is documented in "Navigation actions" directly above this section, so it feels unnecessarily repetitive and prominent.

File: website/docs/icons/usage-guidelines/partials/guidelines/overview.md [hunk 2]
```diff
@@ -119,8 +119,21 @@ Some icons represent common actions within our products.
 
 ### Help actions
 
+- Use `docs-link` when linking to documentation pages and installation guides.
+
+- Use `learn-link` when linking to tutorials.
```
Comment (heatherlarsen) [inline website/docs/icons/usage-guidelines/partials/guidelines/overview.md:124]: Swapped the order to mimic the order in the screenshot.

```suggestion
- Use `learn-link` when linking to tutorials.

- Use `docs-link` when linking to documentation pages and installation guides.
```

**Review by heatherlarsen — APPROVED** [review]
Two minor suggestions; nothing blocking. :shipit:

**Review by MelSumner — APPROVED** [review]
Agree with Heather's suggestions. 👍

**Pull request merged**